Use a random seed, instead of the fixed seed 0

tr7zw opened the pull request:
I recently started using this repo and downloaded https://huggingface.co/MetaIX/Alpaca-30B-Int4 as my model. Quickly noticed that the model always gives the same output for a given input. Turns out, llama models are always run with 0 as seed, instead of -1 for random.

LaaZa commented:
In llama-cpp-python [0 is random](https://github.com/abetlen/llama-cpp-python/blob/2f2ea00a3db0b1dd45af56bfa0bf9464b819982b/llama_cpp/llama.py#LL93C13-L93C45)

tr7zw commented:
Changed it for a draft for now, will check more later. Random seeds are certainly not working correctly(at least with this model). I thought it worked correctly yesterday after I changed it to -1 instead of 0, but right now when asking it "Write a funny joke." it always prints the exact same answer, no matter how often I prompt it/change the parameters.

LaaZa commented:
Might be your other params. Increase the temperature or something.

tr7zw commented:
Hm after reverting to the 0 and cranking up the temperature a bit, now it works. Guess at 0.7 the answers were just so repetitively the same, that I assumed it was broken. Guess don't code while fully exhausted.
